### PR TITLE
Notify PresentationDelegate on all webview uri's

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description
Currently, presentation delegate notifies only the url's opened by the UriService - which is a regression from 2.x (where all successfully opened URI's were notified)
As a solution change the logic to make UiService handle both uri's and url's if the event listener is unable to handle them. Further, notify PresentationDelegate of all uri's opened as part of the interaction.

- Update `handleInAppUri` to `handleInAppUri`
- Remove url validation because we now handle all URI's (handled by UriService)
- Add tests

## Related Issue
(Internal) MOB-20924

## Motivation and Context
Currently, presentation delegate notifies only the url's opened by the UriService which is a regression from 2.x.

Change the logic to make UiService handle both uri's and url's if the event listener is unable to handle them. Further, notify PresentationDelegate of all uri's opened as part of the interaction.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt [hunk 1]
```diff
@@ -108,9 +109,9 @@ internal class InAppMessagePresentable(
                         // So always dismiss the message when a gesture is detected
                         dismiss()
 
-                        // If a gesture mapping exists, the notify the listener about the url associated with the gesture
+                        // If a gesture mapping exists, the notify the listener about the uri associated with the gesture
                         inAppMessage.settings.gestureMap[gesture]?.let { link ->
-                            handleInAppUrl(link)
+                            handleInAppUri(link)
```
Comment (prudrabhat) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt:114]: We do have instrumentation tests that cover part of this logic partially. However, we will need to enhance them if possible to identify if the URI is loaded after the interaction.

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt [hunk 2]
```diff
@@ -169,32 +170,28 @@ internal class InAppMessagePresentable(
...
-            }
-            true
-        } else {
-            false
+    @VisibleForTesting
+    internal fun handleInAppUri(uri: String): Boolean {
+        // First check if the component that created this message is able to handle the uri.
+        // Otherwise check if this URI can be opened by the utility provider via UriOpening service
+        val handled = inAppMessage.eventListener.onUrlLoading(this@InAppMessagePresentable, uri) ||
+            presentationUtilityProvider.openUri(uri)
```
Comment (spoorthipujariadobe) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt:187]: Does 2.x only notify delegate method if `UriService` is able to open the URI? Wondering if there is a case to also notify the delegate about malformed URI

**Review by spoorthipujariadobe — APPROVED** [review]
LGTM, just one small question

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt (hunk 2 above)
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt:187]: In 2.x we notify if the [url is loaded successfully](https://github.com/adobe/aepsdk-core-android/blob/main-v2/code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewClient.java#L162).

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt [hunk 3]
```diff
@@ -169,32 +170,28 @@ internal class InAppMessagePresentable(
         return InAppMessageWebViewClient(
             inAppMessage.settings,
             presentationUtilityProvider
-        ) { url -> handleInAppUrl(url) }
+        ) { url -> handleInAppUri(url) }
```
Comment (praveek) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt:173]: nit: url -> uri as you have renamed in all other places.

**Review by praveek — APPROVED** [review]

**Pull request merged**